### DISCUSSION TITLE
train poly encoder model used `--init-model` model existed in huggingface servers

### DISCUSSION THREAD

**Discussion by salama4ai** [facebookresearch/ParlAI#3610] [Q&A]
i want to fine-tune ploy encoder model while using `--init-model ` of model lives in hugging-face servers,
while to determine `--init-model` in poly-encoder code, need the path to the file that contains 
the model i.e `--init-model zoo:pretrained_transformers/poly_model_huge_reddit/model \` 

but to use hugging-face models they provides code like:
`from transformers import AutoTokenizer, AutoModel`
`base_tokenizer    = AutoTokenizer.from_pretrained("kuisailab/albert-base-arabic")`
`base_model   = AutoModel.from_pretrained("kuisailab/albert-base-arabic")`
 so how to solve this?? how can i get the file path of the model lives in huggingface servers??

**the second question:** 
the original poly encoder model use BERT,  it's ok to use poly encoder with **ALBERT** instead of **BERT** with the same other parameters, right?

**the third one:** 
if i want to use --init-model ((xyz)) so i must use the same tokenizer used in pre-training of ((xyz)) model, right?

**Comment by stephenroller** [comment]
- Both HF and us have string rewrite logic to map these names to URLs. [Here](https://github.com/facebookresearch/ParlAI/blob/a72ef5b2fd0ab03ca3e712f23e5b1811129c01c5/parlai/core/build_data.py#L461) is our logic for it in ParlAI
- Arbitrary HF models aren't going to be compatible with our poly-encoder
- You could switch to ALBERT but you will need to massage the model file and transformer implementation to respect it

**Comment by salama4ai** [comment]
i try to display the data used for pretraining the model `--init-model zoo:pretrained_transformers/poly_model_huge_wikito/model` through `!parlai  display_data --task  ..........` but when i search through the tasks, i didn't find Toronto-books task, and the same for Wikipedia, i found many task that named wekipedia, such as WikiQA , Wikipedia, WikiMovies, wizard_of_wikipedia, which one used for pretraining this model `--init-model zoo:pretrained_transformers/poly_model_huge_wikito/model` ?????? 
shortly i need to see the data that used for pretraining this model `--init-model zoo:pretrained_transformers/poly_model_huge_wikito/model` ??????

**Reply by stephenroller** [reply]
We didn't pretrain the model ourselves. It's the BERT weights converted into ParlAI form.

**Reply by salama4ai** [reply]
![p3](https://user-images.githubusercontent.com/76913472/119183713-c2aca880-ba74-11eb-97d5-6d082b2fd084.png)
i stuck with that error, when i try to use my dataset, i face that error,
example when i try to display dataset, but also when i try to fine tune the model also.
![p1](https://user-images.githubusercontent.com/76913472/119183491-75303b80-ba74-11eb-9032-dd0074d08d64.png)
![p2](https://user-images.githubusercontent.com/76913472/119183509-795c5900-ba74-11eb-8d36-dd4f818d9421.png)
